pxr/base/plug/plugin.cpp

```diff
@@ -253,8 +253,12 @@ PlugPlugin::_Load()
         else {
             string dsoError;
             {
-                TRACE_FUNCTION_SCOPE("dlopen");
+                TRACE_FUNCTION_SCOPE("dlopen");   
+#if defined(ARCH_OS_WINDOWS)
+                _handle = TfDlopen(ArchWindowsPreferredPath(_path).c_str(), ARCH_LIBRARY_ALTERED_SEARCH_PATH, &dsoError);
+#else
                 _handle = TfDlopen(_path.c_str(), ARCH_LIBRARY_NOW, &dsoError);
+#endif
```

Comment:
Instead of having an #ifdef here I think it'd be simpler to have the one function call and just combine `ARCH_LIBRARY_NOW | ARCH_LIBRARY_ALTERED_SEARCH_PATH` since these are already set to platform-specific values in `arch/library.h`.

Reply:
Could you please add a comment about how `ARCH_LIBRARY_ALTERED_SEARCH_PATH` on Windows ensures that .dll dependencies in the same directory as the specified library will be picked up without requiring modifications to the environment?

Reply:
Also, I just noticed `TfDlopen` takes a `std::string` and not a `const char*`, so if you could drop the `.c_str()` while you're here that'd be great